machines/helper.ts

```diff
@@ -3,12 +3,28 @@ import { MachineOptions, EventObject } from 'xstate'
 export const mergeMachineOptions = <TContext, TEvent extends EventObject>(
   machineOptions: Partial<MachineOptions<TContext, TEvent>>[]
 ): Partial<MachineOptions<TContext, TEvent>> => {
-  let masterConfig = {}
+  let masterConfig: Partial<MachineOptions<TContext, TEvent>> = {}
   machineOptions.forEach((config) => {
-    masterConfig = {
-      ...masterConfig,
-      ...config,
-    }
+    ;(masterConfig.actions = {
+      ...masterConfig.actions,
+      ...config.actions,
+    }),
+      (masterConfig.activities = {
+        ...masterConfig.activities,
+        ...config.activities,
+      }),
+      (masterConfig.delays = {
+        ...masterConfig.delays,
+        ...config.delays,
+      }),
+      (masterConfig.guards = {
+        ...masterConfig.guards,
+        ...config.guards,
+      }),
+      (masterConfig.services = {
+        ...masterConfig.services,
+        ...config.services,
+      })
```

Comment:
Why all these parentheses?

Reply:
Not a huge fan of the mutations tho

Reply:
Parentheses seem to be automatically added by formatter/prettier config. Can look into the rule that would allow this.

Reply:
What do you mean by mutations? (the merging syntax?)

Reply:
`masterconfig` wordt hier 'gemuteerd', dat is niet zo mooi